crates/oxc_linter/examples/queries/next_no-before-interactive-script-outside-document/the_rule.yml

```diff
@@ -0,0 +1,294 @@
+name: "next:no-before-interactive-script-outside-document:rule"
+
+query: |
+  query {
+    File {
+      import {
+        from_path @filter(op: "=", value: ["$nextslashscript"])
+        default_import {
+          local_name @tag(name: "default_imported_class_name")
+        }
+      }
+
+      jsx_element {
+        opening_element {
+          name  @filter(op: "=", value: ["%default_imported_class_name"])
+          attribute {
+            name @filter(op: "=", value: ["$strategy"])
+            value_as_constant_string @filter(op: "=", value: ["$beforeInteractive"])
+          }
+          span_: span {
+            start @output
+            end @output
+          }
+        }
+      }
+
+      #### #### #### IGNORE ./src/app or ./app #### #### ####
+
+      # ./app
+      path_part @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+        is_first @filter(op: "=", value: ["$true"])
```

Comment:
do we want a first_path_part helper, considering we have a last_path_part helper?

Reply:
For now I think not, in the future we can add if it's wanted